x-pack/legacy/plugins/siem/public/components/ml/conditional_links/remove_kql_variables.ts

```diff
@@ -18,14 +18,10 @@ export const replacement = (match: string, ...parts: string[]): string => {
   if (parts == null) {
     return '';
   }
-  const operatorsMatched = parts.reduce<string[]>((accum, part) => {
-    if (part != null && operators.includes(part)) {
-      accum = [...accum, part];
-      return accum;
-    } else {
-      return accum;
-    }
-  }, []);
+  const operatorsMatched = parts.reduce<string[]>(
```

Comment:
Up to you but usually what I do when I encounter a pure function[1] that I'm changing is that I review if the author created unit tests specific to the pure function. If they have, I ensure the author covered most of the test cases and corner cases. and if so, everything is good and I don't do anything else.

If I see what looks like a unit test case missing or none at all I usually try to write 1 or 2 tests that exercise the main use case of the function to go with the PR.

In this case, it's really my code and I wrote unit tests that exercise the other functions which exercise this one but not this particular function. That's usually an artifact of either myself or someone else broke down the functions further and probably didn't have time to backfill the other tests. 

My bad here but the reason why I say it's best to do this as a habit even if it's not your code (it is mine in this case) is it will just make you feel better that your change is working as expected and increases our coverage.

But of course it's optional to do it or not, just mentioning ways I think about things as I typically go through the code base.

Reply:
I like the idea of adding relevant tests to make sure the function performs correctly after getting re-worked. I think it will be good for me to acquaint myself with the tests in this PR anyways. I'll add some tests and see what can be covered.